### PR TITLE
feat(connect): add drop support

### PR DESCRIPTION
**Pull request by andrewgazelka**

### CODE REVIEW

**Comment by andrewgazelka** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3345?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3345** <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3345?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#3342** <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3342?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#3344](https://github.com/Eventual-Inc/Daft/pull/3344) <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3344?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3343** <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3343?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#3309](https://github.com/Eventual-Inc/Daft/pull/3309) <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3309?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#3326](https://github.com/Eventual-Inc/Daft/pull/3326) <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3326?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#3341** <a href="https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3341?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by universalmind303 — APPROVED** [review]

File: tests/connect/test_drop.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    # Drop the 'id' column
+    df_dropped = df.drop("id")
+
+    # Verify the drop was successful
+    assert "id" not in df_dropped.columns, "Column 'id' should be dropped"
+    assert len(df_dropped.columns) == len(df.columns) - 1, "Should have one less column after drop"
+
+    # Verify the DataFrame has no columns after dropping all columns"
+    assert len(df_dropped.toPandas().columns) == 0, "DataFrame should have no columns after dropping 'id'"
+    assert df_dropped.count() == df.count(), "Row count should be unchanged after drop"
```
[the remark this answers is not in the document]
Comment (andrewgazelka) [reply tests/connect/test_drop.py:17]: Well, in our case, I don't think we have `.to_pydict` in spark, right? We have `createDataFrame`, and that's a feat that needs to be added/merged

right now it is still a WIP

`andrew/connect-create-dataframe`

[the remark this answers is not in the document]
Comment (andrewgazelka) [reply tests/connect/test_drop.py:17]: ok...

[the remark this answers is not in the document]
Comment (universalmind303) [reply tests/connect/test_drop.py:17]: `createDataFrame` is row oriented, so you can't pass in the same data type, but you can go through arrow. 

```py
tbl = daft.from_pydict({...}).to_arrow()
df = createDataFrame(tbl)
```

so we could probably just create some util functions that's used throughout the connect codebase

```py
def make_spark_df(data):
  tbl = daft.from_pydict({...}).to_arrow()
  return createDataFrame(tbl)
```

**Comment by andrewgazelka** [comment]
### Merge activity

* **Dec 11, 4:57 PM EST**: [Graphite](https://app.graphite.dev/github/pr/Eventual-Inc/Daft/3345) disabled "merge when ready" on this PR due to: a merge conflict with the target branch; resolve the conflict and try again..

**Pull request merged**